src/character.cpp

```diff
@@ -1936,6 +1936,7 @@ bool Character::pour_into( item &container, item &liquid )
         auto qty = std::min( liquid.charges, container.ammo_capacity() - container.ammo_remaining() );
         liquid.charges -= qty;
         container.ammo_set( liquid.typeId(), container.ammo_remaining() + qty );
+        container.on_contents_changed();
```

Comment:
A callback is a well designed solution but how robust is this considering the poor (but improving) encapsulation of `contents`?